static/js/components/Group.jsx

```diff
@@ -258,27 +259,24 @@ const Group = () => {
   };
 
   const { id } = useParams();
-  const loadedId = useSelector((state) => state.group?.id);
-
-  useEffect(() => {
-    const fetchGroup = async () => {
-      const data = await dispatch(groupActions.fetchGroup(id));
-      if (data.status === "error") {
-        setGroupLoadError(data.message);
-      }
-    };
-    if (id !== loadedId) {
-      fetchGroup();
-    }
-  }, [id, loadedId, dispatch]);
 
   const group = useSelector((state) => state.group);
   const currentUser = useSelector((state) => state.profile);
   const { invitationsEnabled } = useSelector((state) => state.sysInfo);
-
-  // fetch streams:
   const streams = useSelector((state) => state.streams);
 
+  useEffect(() => {
+    const fetchGroup = async (groupID) => {
+      const result = await dispatch(groupActions.fetchGroup(groupID));
+      if (result.status === "error") {
+        setGroupLoadError(result.message);
+      }
+    };
+    if (String(id) !== String(group?.id)) {
+      fetchGroup(id);
```

Comment:
If id is invalid, won't this cause an infinite loop?

I know we've had to make special arrangements in other similar patterns to avoid that.

Reply:
Note this logic is not new to this PR, just got moved around

Reply:
I'll add a fix to this PR though (dataFetched component state that gets set to true after initial fetch, as we've done elsewhere)

Reply:
Done